hack/tools/release/notes/main.go

```diff
@@ -21,414 +21,102 @@ limitations under the License.
-		"ipam":                              "IPAM",
-		"provider/bootstrap-kubeadm":        "CAPBK",
-		"provider/infrastructure-in-memory": "CAPIM",
-		"provider/core":                     "Core",
-		"runtime-sdk":                       "Runtime SDK",
-		"ci":                                "CI",
-		"machinehealthcheck":                "MachineHealthCheck",
-		"clusterctl":                        "clusterctl", // Preserve lowercase
-		"util":                              "util",       // Preserve lowercase
-		"community-meeting":                 "Community meeting",
-	}
-
-	releaseBackportMarker = regexp.MustCompile(`(?m)^\[release-\d\.\d\]\s*`)
-)
-
 func main() {
-	flag.Parse()
-	os.Exit(run())
-}
-
-func lastTag() string {
-	if fromTag != nil && *fromTag != "" {
-		return *fromTag
-	}
-	cmd := exec.Command("git", "describe", "--tags", "--abbrev=0")
-	out, err := cmd.Output()
-	if err != nil {
-		return firstCommit()
+	cmd := newNotesCmd()
+	if err := cmd.run(); err != nil {
+		log.Fatal(err)
 	}
-	return string(bytes.TrimSpace(out))
 }
 
-func firstCommit() string {
-	cmd := exec.Command("git", "rev-list", "--max-parents=0", "HEAD")
-	out, err := cmd.Output()
-	if err != nil {
-		return "UNKNOWN"
-	}
-	return string(bytes.TrimSpace(out))
+type notesCmdConfig struct {
+	repo                        string
+	fromRef                     string
+	toRef                       string
+	newTag                      string
+	branch                      string
+	prefixAreaLabel             bool
+	preReleaseVersion           bool
+	deprecation                 bool
+	addKubernetesVersionSupport bool
 }
 
-// Since git doesn't include the last day in rev-list we want to increase 1 day to include it in the interval.
-func increaseDateByOneDay(date string) (string, error) {
-	layout := "2006-01-02"
-	datetime, err := time.Parse(layout, date)
-	if err != nil {
-		return "", err
-	}
-	datetime = datetime.Add(time.Hour * 24)
-	return datetime.Format(layout), nil
-}
-
-const (
-	missingAreaLabelPrefix   = "MISSING_AREA"
-	areaLabelPrefix          = "area/"
-	multipleAreaLabelsPrefix = "MULTIPLE_AREAS["
-	documentationAreaLabel   = "Documentation"
-)
-
-type githubPullRequest struct {
-	Labels []githubLabel `json:"labels"`
-}
-
-type githubLabel struct {
-	Name string `json:"name"`
-}
-
-func getAreaLabel(merge string) (string, error) {
-	// Get pr id from merge commit
-	prID := strings.Replace(strings.TrimSpace(strings.Split(merge, " ")[3]), "#", "", -1)
-
-	cmd := exec.Command("gh", "api", fmt.Sprintf("repos/%s/pulls/%s", *repo, prID)) //nolint:gosec
-
-	out, err := cmd.CombinedOutput()
-	if err != nil {
-		return "", fmt.Errorf("%s: %v", string(out), err)
-	}
+func readCmdConfig() *notesCmdConfig {
+	config := &notesCmdConfig{}
 
-	pr := &githubPullRequest{}
-	if err := json.Unmarshal(out, pr); err != nil {
-		return "", err
-	}
+	flag.StringVar(&config.repo, "repository", "kubernetes-sigs/cluster-api", "The repo to run the tool from.")
```

Comment:
nit: It might be helpful to specify the format of the repo

Reply:
isn't the default enough? or what are you looking for?

Reply:
Ah that's my bad. I forgot the default will show up with -h